### DISCUSSION TITLE
Github Pages with single custom (sub)domain for multiple repositories

### DISCUSSION THREAD

**Discussion by tenbits** [community/community#22433] [Other Feature Feedback, Questions, & Ideas]
<p>I have multiple Github Repositories. By all of them <code>pages</code> are enabled, and the projects are accessible via paths for example:</p>
<ul>
<li>
<code>username.github.io/project1</code> ,</li>
<li>
<code>username.github.io/project2</code> ,</li>
<li><code>...</code></li>
</ul>
<p>Now I’m trying to create a custom subdomain <code>docs.foo.tld</code> and to have all my projects <code>gh-pages</code> to be accessible via</p>
<ul>
<li><code>docs.foo.tld/project1</code></li>
<li><code>docs.foo.tld/project2</code></li>
<li><code>...</code></li>
</ul>
<p>I have created a <code>CNAME</code> entry <code>docs.foo.tld</code> to <code>username.github.io</code></p>
<p>And when I’m adding this custom domain ( <code>docs.foo.tld</code> ) in <code>project1</code> repository settings on github, the <code>pages</code> are directly accessible via <code>docs.foo.tld</code> , not via <code>docs.foo.tld/project1</code></p>
<p>How I use all this projects in the path segments <code>/project1</code> , <code>/project2</code> when using same custom subdomain for all of them.</p>
<blockquote>
<p>When I try to add this custom subdomain to <code>project2</code> , I get the exception as the <code>CNAME</code> is already taken.</p>
</blockquote>

**Comment by jsoref** [comment]
<p>I don’t think github pages can do what you want the way you’re trying to do it.</p>
<p>You could just merge all the docs into a single repository.</p>
<p>Or you could use your own hosting service and have it do this for you.</p>

**Comment by yoannchaudet** [comment]
<p>Hello and welcome here!</p>
<blockquote>
<p>I have created a <code>CNAME</code> entry <code>docs.foo.tld</code> to <code>username.github.io</code></p>
</blockquote>
<p>This should be sufficient. The special repository named <code>&lt;your username&gt;.github.io</code> is what we call your user site. If you set a custom domain on that repository, all your other Pages sites (project sites) will automatically route on that new domain. Only one repository at a time can be associated with a given custom domain (we enforce the unicity).</p>
<p>You can learn more about the different type of Pages sites here: <a href="https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites" class="inline-onebox">About GitHub Pages - GitHub Docs</a>.</p>

**Comment by Randall-Scharpf** [comment]
<p>Hello everyone! I’m a few months late to the thread, but I’m attempting the same configuration, and somehow, following your steps leaves me with a 404, in a way I cannot comprehend. I’ll provide the details of what I’ve done so far below, what I’m trying to do (which will be pretty apparent from what I’ve done), and some final questions for things I might want to do sometime in the future.</p>
<p>First, here’s what I have. (I’ve removed specific information like my username and project name from the screenshots even though you could literally just google them so that anyone else with this issue can more easily refer to this thread)</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/4/945bbe92d8e6ad226bbdd026efd74c87b928630e.png" data-download-href="/uploads/short-url/lare5bXDy9zN5mMZL3alF91kEP4.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/43557842/181070878-4501cd29-675c-42cd-adb3-f47e4462fec4.png" alt="image" data-base62-sha1="lare5bXDy9zN5mMZL3alF91kEP4" width="419" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/4/945bbe92d8e6ad226bbdd026efd74c87b928630e_2_419x500.png, https://aws1.discourse-cdn.com/github/original/3X/9/4/945bbe92d8e6ad226bbdd026efd74c87b928630e.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/4/945bbe92d8e6ad226bbdd026efd74c87b928630e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/4/945bbe92d8e6ad226bbdd026efd74c87b928630e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">492×587 56.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>What am I doing wrong?</p>
<p>As a follow-up question, I might, in the future, think about adding new subdomains containing multiple projects to my github (say, <code>python.mywebsite.com/MyPythonProject</code> ), and I don’t see any way to do that under this configuration, even if it works. I am aware that I could merge the website portions of ALL my java projects into one repository and the website portions of ALL my theoretical python projects into one repository, but this would complicate my build tooling every time I do an update, so if it’s possible, I’d prefer to have this set up through github pages.<br>
If this is impossible, I’m curious as to whether that’s just something not implemented or if it’s an actual technical limitation with how GitHub Pages is implemented, since if it’s a technical limitation, I might have to mind related gotchas while workarounding.</p>
<p>Thanks!<br>
-Randall</p>

**Comment by tenbits** [comment]
<p>From the screenshot: you haven’t set your custom domain in GitHub Pages: <code>https://github.com/USERNAME/USERNAME.github.io/settings/pages</code> there is a section: “Custom Domain” - so there you have to provide your <code>java.example.com</code>.</p>

**Comment by Randall-Scharpf** [comment]
<p>Updating the repository cname through the pages settings puts my website at <code>java.mywebsite.com</code>, not <code>java.mywebsite.com/jMyProject</code>.</p>

**Comment by port19x** [comment]
git submodules + symlinks can achieve this

I have a folder `external` at the root of the blog repos directory.
I put the git submodule in said folder.
My project `dvd-animation` has it's html at `/resources/public/`
I then create a smylink: `ln -s external/dvd/animation/resources/public docs/dvd-animation`
